### ISSUE TITLE
AsyncContext::wait() is not actually waiting

### ISSUE THREAD

**Issue by gdesmott** [actix/actix#548]
`AsyncContext::wait()` is [documented](https://docs.rs/actix/latest/actix/trait.AsyncContext.html#tymethod.wait) as :

```
Spawns a future into the context, waiting for it to resolve.
```

But according to my tests `wait()` is actually not blocking and return right away, instead of waiting for the future to be done.

This is pretty confusing, it would be good to clarify its exact behavior in the documentation.

**Comment by Sytten** [comment]
AFAIK it means that the actor will wait on that future before making anymore progress (like reading new messages or polling children futures).

**Comment by gdesmott** [comment]
Indeed, that seems to match was I observed and my understanding of the code.

I think it's worth to make the documentation clearer about the exact semantic of this API as I spent quite some time figuring it out.

**Comment by Sruhvx-jpg** [comment]
hey @gdesmott  i think the docs were updated or maybe had this exact behavior documented


`
This stops processing any incoming events until the future resolves.
`

If you think this is explicit enough, we can mark this issue as resolved
